lexicon/providers/yandexcloud.py

```diff
@@ -0,0 +1,307 @@
+                for entry in record["data"]:
+                    # artificially generate ID for deletion
+                    record_id = f"{record['type']}${self._relative_name(record['name'])}${entry}"
+                    records.append({
+                        "id": record_id,
+                        "name": self._full_name(record["name"]),
+                        "type": record["type"],
+                        "ttl": record["ttl"],
+                        "content": entry,
+                    })
+
+        if content:
+            records = [record for record in records if record["content"].lower() == content.lower()]
+
+        LOGGER.debug("list_records: %s", records)
+        return records
+
+    def _get_record(self, rtype: str, name: str) -> Dict:
+        """Gets a single record by rtype and name.
+        Throws HTTPError with status code 404 if record not found.
+        https://cloud.yandex.com/en/docs/dns/api-ref/DnsZone/getRecordSet"""
+
+        if name is not None:
+            name = self._full_name(name) + "."
+
+        url = f"/zones/{self.domain_id}:getRecordSet"
+        query_params = {"name": name, "type": rtype}
+        # name should be in full form, transform it in case it's not
+
+        payload: Dict = self._get(url, query_params)
+
+        return {
+            "name": payload["name"],
+            "type": payload["type"],
+            "ttl": payload["ttl"],
+            "data": payload["data"],
+        }
+
+    def _create_record(self, rtype, name, content):
+        """Create a record. Return True if record was created, or already the same precise record exists,
+         False if record already exists or there is another problem.
+         All params are required.
+        """
+
+        # name should be in full form, transform it in case it's not
+        if name is not None:
+            name = self._full_name(name) + "."
+
+        # get existing content
+        data = [content]
+        try:
+            record = self._get_record(rtype, name)
+            data = record["data"]
+            # add new content entry only if it's not yet present
+            if content not in record["data"]:
+                data.append(content)
+        except requests.exceptions.HTTPError as error:
+            if error.response.status_code != 404:
+                raise error
+
+        payload = self._post(f"/zones/{self.domain_id}/upsertRecordSets", {
+            "replacements": [
+                {
+                    "name": name,
+                    "type": rtype,
+                    "ttl": self._get_lexicon_option("ttl") or self.default_ttl,
+                    "data": data,
+                }
+            ]
+        })
+
+        return self._check_request_success(payload, "create_record")
+
+    def _update_record(self, identifier=None, rtype=None, name=None, content=None):
+        """Update existing record. Identifier is not used for Yandex Cloud."""
```

Comment:
Usually `_update_record` covers two scenarios:
* `identifier` is passed, and the rtype/name/content are provided as inputs for the updated record
* `rtype`, `name` and `content` are passed, and `rtype`/`name` are considered as the lookup parameters, and `content` the input for the updated record.

Arguably the second situation is not clear and parameters usage is not aligned with the first case. I would like to make it clear in the future.

Anyway, the support of identifier should be added. Depending on if you follow my advice for id generation, or keep a decodable id, you should add this logic first to fetch the record, then use its content to actually update it.